.pre-commit-config.yaml

```diff
@@ -12,6 +12,13 @@ repos:
         entry: Forbid symlinks
         language: fail
         types: [symlink]
+  - repo: local
+    hooks:
+      - id: check-line-endings
+        name: check for CRLF line endings
```

Comment:
Isn't there a pre-commit hook in pre-commit-hooks that does this already?

Reply:
Not that I could find 🤷‍♂️

Reply:
I see https://github.com/pre-commit/pre-commit-hooks#mixed-line-ending

Does `--fix=lf` only rewrite the file if it has mixed line endings?

Reply:
Ooh interesting; missed that.  I had assumed mixed was just checking for mixed.  I wonder how they avoid messing with binary files; the advantage of our approach is that we explicitly mark binary files, and we never actually touch line endings, just flag them, in case someone checks something in with a new file ending not in our list

Reply:
Does our approach seem ok here @auscompgeek?  I think I like _not_ running as a fixer, as we do here, so that we don't inadvertently break someone's binary on checkin if they have commit hooks set up

Reply:
Ok if no more objections in the next 24 hrs then I'm merging this one in @auscompgeek

Reply:
> I had assumed mixed was just checking for mixed.

I did too given its name. But then I took a look behind the curtain. 

> I wonder how they avoid messing with binary files

It filters by type:

https://github.com/pre-commit/pre-commit-hooks/blob/b0f11fe246ae49c7a3791009753d2148b97cb677/.pre-commit-hooks.yaml#L165-L170

`types` are documented here: https://pre-commit.com/#filtering-files-with-types

Reply:
yeah I think I'm maybe too paranoid for "looks like a binary file" 😅

Reply:
I mean, if it's [good enough for libmagic](https://github.com/pre-commit/identify/blob/d70d842caf3d9b06560de4db74a98260236f592a/identify/identify.py#L121), surely it's good enough for us? I can't imagine anyone checking in any binary files where the first 1KB happens to not have any control characters

Reply:
FWIW https://github.com/Lucas-C/pre-commit-hooks looks like it has a non-fixing check too

Reply:
🤝